templates/cube/study/index.md

```diff
@@ -19,4 +19,22 @@ For additional practise, enroll in Canonical’s Study Labs. The labs feature vi
 
 A single purchase grants 90 days of access to the full set of Study Labs.
 
-<a class="p-button--positive">Purchase Study Lab access</a>
\ No newline at end of file
+<a class="p-button--positive js-study-button"><i class="p-icon--spinner u-animation--spin is-light"></i></a>
+
+<script>
+  const button = document.querySelector(".js-study-button");
+  fetch("study/labs")
+    .then((response) => {
+      if (!response.ok) {
+        throw new Error('Response was not ok');
+      }
+      return response.json()
+    })
+    .then((json) => {
+      button.href = json.redirect_url;
+      button.text = json.text;
+    })
+    .catch(() => {
+      button.remove()
```

Comment:
Could we provide a more useful thing to the user if this fails unexpectedly. For example, disable the button and change text to error message?